style/content.rst

```diff
@@ -0,0 +1,143 @@
+===============
+ Content Guide
+===============
+
+Principles
+==========
+
+* Use your judgement about following these recommendations
+  
+  * OK to consider ROI
+
+* Strive for consistency across components
+
+  * Especially newer specs
+  * Component must be consistent with itself
+
+* Target audience
+
+  * implementors
+  * sophisticated users who will be providing feedback
+
+    * assume final product will have separate programming guide with
+      tutorial-level information
+
+* Spec must be publishable as part of unified doc and as standalone
+  describing a single component
+
+  * when published as a standalone, it will contain some sections that
+    are normally at top-level. e.g. bibliography
+  
+Structure of Document
+=====================
+
+* Top-level spec contains sections for information common to all
+  libraries
+
+  * Conventions
+
+    * C++ standard compliance 
+    * Basic scalar data types
+    * Exceptions
+    * Usm pointers
+    * Guidelines (or just guidelines that include all other conventions?)
+    * Namespaces (dedicated table of the namespaces?)
+
+  * Math notations
+  * Glossary: terms common to multiple components. e.g. SYCL, DPC++, USM
+  * Bibliography
+
+* Component-level sections for information specific to the component
+  
+  * Introduction: defines the scope of the SW
+  * Glossary: defines terms used to describe specs
+  * Math notations
+  * Namespace-related information: e.g., the first level namespace,
+    the lib introduces additional namespaces based on some internal
+    logic
+  * header files
+  * SW API usage with examples demonstrating the API of the lib
```

Comment:
There will be at least 1 API for "get_spec_version" that should be common for all libs i think.   So user will have same way for getting it regardless of product.   

We ether can follow approach as we have with other design decisions there we align on API/design but have own implementations.  Or move such things to common place. 

The main difference is that for spec version we should provide same API and same behavior across libs.

Reply:
I will add some requirement to the common section, but the actual API needs to be specified in each component.

Reply:
'Conventions' would contain the list of the cross-library consistency topics incl spec version and version format. Respective proposal on the spec version suggests have it in the specs but gives the flexibility how to implement it. Whether to have the lib version in specs or not - up to the library